core/src/main/java/org/commonjava/indy/core/content/DefaultDownloadManager.java

```diff
@@ -473,6 +473,10 @@ private Transfer retrieve( final ArtifactStore store, final String path, final b
             {
                 target = transfers.retrieve( res, suppressFailures, eventMetadata );
             }
+            else if ( nfc.isMissing( res ) )
```

Comment:
Why not put this as the first condition in the set? We don't want to attempt retrieval if it's in the NFC, right? I mean, the TransportManager should also be checking the NFC, so I guess it's not critical, but it's a little confusing to read.